File: 백준/Silver/21921. 블로그/블로그.py
from sys import stdin

n, x = map(int, stdin.readline().split())
visit_counts = list(map(int, stdin.readline().split()))

max_visit, count = sum(visit_counts[:x]), 1
current_visit = max_visit

for i in range(x, len(visit_counts)):
    current_visit = current_visit - visit_counts[i - x] + visit_counts[i]
    if current_visit > max_visit:
        max_visit, count = current_visit, 0
    elif current_visit < max_visit:
        continue
    count += 1
        
if max_visit == 0:
    print("SAD")
else:
    print(max_visit, count, sep='\n')

File: 백준/Silver/21921. 블로그/블로그.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken()), x = Integer.parseInt(st.nextToken());
        int visits[] = new int[n];
        st = new StringTokenizer(br.readLine());
        for (int i = 0; st.hasMoreTokens() ; i++) {
            visits[i] = Integer.parseInt(st.nextToken());
        }
        int sum = 0;
        for (int i = 0 ; i < x ; i++) {
            sum += visits[i];
        }
        int max = sum, count = 1;
        for (int i = x ; i < n ; i++) {
            sum += visits[i] - visits[i-x];
            if (sum > max) {
                max = sum;
                count = 1;
            } else if (sum == max) {
                count++;
            }
        }

        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        if (max == 0) {
            bw.write("SAD");
        } else {
            bw.write(String.valueOf(max) + "\n" + String.valueOf(count));
        }
        bw.flush();
    }
}